app/gui/src/model/suggestion_database.rs

```diff
@@ -152,14 +265,28 @@ impl SuggestionDatabase {
                 },
                 entry::Update::Remove { id } => {
                     let removed = entries.remove(&id);
-                    if removed.is_none() {
-                        error!(self.logger, "Received Remove event for nonexistent id: {id}");
+                    match removed {
+                        Some(entry) => {
+                            let qualified_name = entry.qualified_name();
+                            qn_to_id_map.warn_if_absent_and_remove(&qualified_name);
```

Comment:
TBH I don't see any reason these are 2 lines. With expression `qn_to_id_map.warn_if_absent_and_remove(&entry.qualified_name())` the code would probably be shorter, wouldn't it?